tests/unit/AbortController.ts

```diff
@@ -0,0 +1,94 @@
+import { AbortController, AbortSignal, ShimAbortController, ShimAbortSignal } from '../../src/AbortController';
+import global from '../../src/global';
+
+let controller: AbortController;
+let signal: AbortSignal;
+
+import has from '../../src/support/has';
+
+const { registerSuite } = intern.getInterface('object');
+const { assert } = intern.getPlugin('chai');
+
+registerSuite('AbortController and AbortSignal', {
+	beforeEach() {
+		controller = new ShimAbortController();
+		signal = controller.signal;
+	},
+	tests: {
+		'native AbortController'() {
+			if (!has('abort-controller')) {
+				this.skip('checking native implementation');
+			}
+			assert.include(
+				ShimAbortController.toString(),
+				'native',
+				'native implementation should represent native code'
+			);
+		},
+
+		'native AbortSignal'() {
+			if (!has('abort-signal')) {
+				this.skip('checking native implementation');
+			}
+			assert.include(ShimAbortSignal.toString(), 'native', 'native implementation should represent native code');
+		},
+
+		'AbortController defaults'() {
+			assert.isFunction(controller.abort, 'should provide an `abort` method');
+			assert.isDefined(controller.signal, 'should provide a signal property');
+		},
+
+		'AbortSignal defaults'() {
+			assert.isFalse(signal.aborted);
+			assert.notExists(signal.onabort);
+		},
+
+		'abort event'() {
+			const dfd = this.async();
+			signal.addEventListener(
+				'abort',
+				dfd.callback((event: Event) => {
+					assert.strictEqual(event.type, 'abort');
+					assert.isTrue(signal.aborted);
+				})
+			);
+			controller.abort();
+		},
+
+		'onabort method callback'() {
+			const dfd = this.async();
+			signal.onabort = dfd.callback((event: Event) => {
+				assert.strictEqual(event.type, 'abort');
+			});
+			controller.abort();
+		},
+
+		'remove event listener'() {
+			let called = false;
+			const callback = () => (called = true);
+			signal.addEventListener('abort', callback);
+			signal.removeEventListener('abort', callback);
+			controller.abort();
+			assert.isFalse(called, 'callback should not have been called');
+		},
+
+		'dispatch event'() {
+			return new Promise((resolve) => {
+				let called = 0;
+				const expectedCalls = 2;
+				const createEvent = (type: string) => (global.Event ? new Event(type) : ({ type } as Event));
```

Comment:
would document.createEvent('Event'); work here in place of the constructor for IE11?